### PR TITLE
br: modify fed restore status and ensure init job deleted

### LINKED ISSUE #5076: ebs-br: job name exceeds 63 characters limit [keep it as a known limitation]
## Bug Report

**What version of Kubernetes are you using?**

Nightly

**What version of TiDB Operator are you using?**

Nightly

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

Nightly

**What's the status of the TiDB cluster pods?**

Nightly

**What did you do?**

1. Create volume backup schedule CRD and apply it

**What did you expect to see?**
1. The schedule can run successfully

**What did you see instead?**

img_v2_1888c3d8-be49-4c87-9c62-788adc39685g

### LINKED ISSUE #5088: Volume backup status should be displayed as Complete, with the first letter capitalized [by 7/7]
## Bug Report

**What version of Kubernetes are you using?**

**What version of TiDB Operator are you using?**
wangle1321/tidb-operator:202305261650

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**
gp3

**What's the status of the TiDB cluster pods?**

**What did you do?**
1. create cluster(cross k8s)
2. create volumebackup

**What did you expect to see?**
1. Volume backup status should be displayed as "Complete"

**What did you see instead?**
1. Volume backup status should be displayed as "complete"
image

### LINKED ISSUE #5141: Volume backup failed but the initializing job was still running
## Bug Report

**What version of Kubernetes are you using?**

**What version of TiDB Operator are you using?**

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

**What did you do?**

create a VolumeBackup CR with the backup storage that has backup meta file. So the VolumeBackup will fail because the backup meta file already exists.

**What did you expect to see?**

VolumeBackup fails, and no backup job running.

**What did you see instead?**

VolumeBackup fails, but the initializing job is still running. aws snapshot will  running，which will cause next backup time cost long time(the next snapshot backup will begin after pre snapshot task completed)
2166f380-40f7-4c15-9b39-827c254b4705
j4EO7JA6Df

### PR DESCRIPTION
**Pull request by WangLe1321**
### What problem does this PR solve?

1. After the volume backup is failed, ensure the initializing job is stopped
Close #5141

2. Modify VolumeRestore condition started with uppercase letter.
Close #5088

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/helpers.go [hunk 1]
```diff
@@ -14,13 +14,21 @@
 package v1alpha1
 
 import (
+	// #nosec
```
Comment (BornChanger) [inline pkg/apis/pingcap/v1alpha1/helpers.go:17]: Please add a little bit more on why use `nosec` here.

File: pkg/apis/pingcap/v1alpha1/helpers.go [hunk 2]
```diff
@@ -14,13 +14,21 @@
...
 	"hash/fnv"
 
 	"k8s.io/apimachinery/pkg/util/rand"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
+const (
+	// See https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names
+	labelLengthLimit = 63
+	hashSize         = 8
```
Comment (BornChanger) [inline pkg/apis/pingcap/v1alpha1/helpers.go:29]:
```suggestion
	LabelLengthLimit = 63
	HashSize         = 8
```

File: pkg/backup/backup/backup_manager.go [hunk 3]
```diff
@@ -84,13 +84,22 @@ func (bm *backupManager) syncBackupJob(backup *v1alpha1.Backup) error {
 	var err error
 
 	if backup.Spec.Mode == v1alpha1.BackupModeVolumeSnapshot {
-		// check volume backup initialize job, we should ensure the job is existed during volume backup
-		if err = bm.checkVolumeBackupInitializeJobExisted(backup); err != nil {
+		if v1alpha1.IsBackupFailed(backup) || v1alpha1.IsVolumeBackupFailed(backup) {
+			// if volume backup failed, we should delete initialize job to prevent GC and pd schedule blocked
```
Comment (BornChanger) [inline pkg/backup/backup/backup_manager.go:88]:
```suggestion
			// if volume backup failed, we should delete initialize job to resume GC and pd schedule
```

File: pkg/backup/backup/backup_manager.go [hunk 4]
```diff
@@ -983,6 +992,27 @@ func (bm *backupManager) teardownVolumeBackup(backup *v1alpha1.Backup) (err erro
...
 
+func (bm *backupManager) deleteVolumeBackupInitializeJob(backup *v1alpha1.Backup) error {
+	ns := backup.GetNamespace()
+	name := backup.GetName()
+	backupInitializeJobName := backup.GetVolumeBackupInitializeJobName()
+
+	backupInitializeJob, err := bm.deps.JobLister.Jobs(ns).Get(backupInitializeJobName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil
```
Comment (BornChanger) [inline pkg/backup/backup/backup_manager.go:1003]: add a warning level log here

Comment (WangLe1321) [reply pkg/backup/backup/backup_manager.go:1003]: Job not found is what we expect. It indicates the job already deleted, so I think we don't need warning log here.

File: pkg/backup/backup/backup_manager.go (hunk 3 above)
Comment (WangLe1321) [reply pkg/backup/backup/backup_manager.go]: done

File: pkg/backup/backup/backup_manager.go (hunk 4 above)
Comment (WangLe1321) [reply pkg/backup/backup/backup_manager.go:1004]: The init job not found is what we expect, so I add a info log here.

**Review by csuzhangxc — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**